dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/tracking/AndroidXFragmentLifecycleCallbacks.kt

```diff
@@ -51,17 +53,29 @@ internal class AndroidXFragmentLifecycleCallbacks(
 
     override fun onFragmentResumed(fm: FragmentManager, f: Fragment) {
         super.onFragmentResumed(fm, f)
-        GlobalRum.get()
-            .startView(
-                f,
-                f.resolveViewName(),
-                argumentsProvider(f)
-            )
+        validateFragmentAndExecute(f) {
+            GlobalRum.get()
+                .startView(
+                    it,
+                    it.resolveViewName(),
+                    argumentsProvider(it)
+                )
+        }
     }
 
     override fun onFragmentPaused(fm: FragmentManager, f: Fragment) {
         super.onFragmentPaused(fm, f)
-        GlobalRum.get().stopView(f)
+        validateFragmentAndExecute(f) { GlobalRum.get().stopView(it) }
+    }
+
+    // endregion
+
+    // region internal
+
+    private fun validateFragmentAndExecute(f: Fragment, operation: (Fragment) -> Unit) {
```

Comment:
I'm not a fan of making such functions, as it doesn't really help reading. Maybe it's the name, but reading `validateFragmentAndExecute(f) { … }` makes me thing the validation actually does something (that is has some side effect). 

Maybe renaming it `runIfValid` or something like that can help with the readability, but since you're just wrapping an if, I don't see why we can't use that. If the boolean condition was "trickier", maybe but here it doesn't add any value I think.

If you really want to reuse code, which i totally get, then you can make an extension function on a predicate, and use it in all the ViewTrackingStrategies

```kotlin

fun <T> Predicate<T>.runIfValid(t : T, operation : (T) -> Unit) {
    if (accept(t)) {
        operation(t)
    }
}

// …

predicate.runIfValid(f) { GlobalRum.get().stopView(it) }
```